### PR TITLE
Fixed: null reference instead of assert on failed reflection in CookieExtensions

### PR DESCRIPTION
**Pull request by sstronin**
There are asserts that never could be reached because null references occur first on CreateDelegate calls.

### CODE REVIEW

**Comment by stephentoub** [comment]
> There are asserts that never could be reached because null references occur first on CreateDelegate calls.

Can you elaborate? Null dereference from where? If GetMethod returns null, CreateDelegate won't be called because it's using "?." rather than ".".

**Comment by stephentoub** [comment]
Oh, nevermind, I was reading your changed code rather than the original :)

File: src/libraries/System.Net.HttpListener/src/System/Net/Windows/CookieExtensions.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ public static string ToServerString(this Cookie cookie)
 
         public static Cookie Clone(this Cookie cookie)
         {
-            s_cloneFunc ??= (Func<Cookie, Cookie>)typeof(Cookie).GetMethod("Clone", BindingFlags.Instance | BindingFlags.NonPublic)!.CreateDelegate(typeof(Func<Cookie, Cookie>));
+            s_cloneFunc ??= (Func<Cookie, Cookie>)typeof(Cookie).GetMethod("Clone", BindingFlags.Instance | BindingFlags.NonPublic)?.CreateDelegate(typeof(Func<Cookie, Cookie>))!;
```
Comment (hughbe) [inline src/libraries/System.Net.HttpListener/src/System/Net/Windows/CookieExtensions.cs:26]: why would these methods not exist?

Comment (sstronin) [reply src/libraries/System.Net.HttpListener/src/System/Net/Windows/CookieExtensions.cs:26]: If there were any expected circumstances for that, there should be proper error handling, not just asserts. But asserts are here so they deserve a chance to hit.

**Review by MihaZupan — APPROVED** [review]
Seems fine. If something does go wrong we'll still throw NRE when we go to invoke the delegate, we'll just get better info in debug.

Just to confirm, this was based solely on reading the code / tooling analysis, and not an actual issue you ran into, right?

**Comment by sstronin** [comment]
Yes, this was found by a code analyzer.

**Pull request merged**